CBLClient/Apps/CBLTestServer-iOS/CBLTestServer-iOS/Server/ReplicatorRequestHandler.swift

```diff
@@ -0,0 +1,104 @@
+    
+    public func handleRequest(method: String, args: Args) throws -> Any? {
+        switch method {
+        /////////////////
+        // Replication //
+        /////////////////
+        case "replicator_create":
+            let config: ReplicatorConfiguration? = args.get(name: "config")
+            return Replicator(config: config!)
+            
+        case "replicator_start":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            replication_obj.start()
+
+        case "replicator_stop":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            replication_obj.stop()
+
+        case "replicator_status":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            let value = replication_obj.status.stringify()
+            print("displaying replication status \(value)")
+            return value
+
+        case "replicator_config":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            return replication_obj.config
+
+        case "replicator_getActivityLevel":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            return String(describing: replication_obj.status.activity)
+
+        case "replicator_getCompleted":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            return replication_obj.status.progress.completed
+
+        case "replicator_getTotal":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            return replication_obj.status.progress.total
+
+        case "replicator_getError":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            return String(describing: replication_obj.status.error)
+
+        case "replicator_addChangeListener":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            let changeListener = MyReplicationChangeListener()
+            let listenerToken = replication_obj.addChangeListener(changeListener.listener)
+            changeListener.listenerToken = listenerToken
+            return changeListener
+
+        case "replicator_removeChangeListener":
+            let replication_obj: Replicator = args.get(name: "replicator")!
+            let changeListener : MyReplicationChangeListener = (args.get(name: "changeListener"))!
+            replication_obj.removeChangeListener(withToken: changeListener.listenerToken!)
+
+        case "replicator_changeListenerChangesCount":
+            let changeListener: MyReplicationChangeListener = (args.get(name: "changeListener"))!
+            return changeListener.getChanges().count
+
+        case "replicator_changeListenerGetChanges":
+            let changeListener: MyReplicationChangeListener = (args.get(name: "changeListener"))!
+            // let index: Int = (args.get(name: "index"))!
+            return changeListener.getChanges().description
```

Comment:
we can make index as optional . If index is passed , then changes will have limit otherwise it should get all changes. Not sure is differrence with and without description.

Reply:
Removed it. MyReplicationChangeListener doesn't seem to take an index.